src/NodeVisitor/DynamicLoader.php

```diff
@@ -17,16 +17,146 @@ class DynamicLoader extends NodeVisitorAbstract
 {
     public $definitionCollector;
     public $prettyPrinter;
+    public $definitionResolver;
+    
+    private $collectAutoload;
```

Comment:
To make code cleaner we might need to put the file name as an argument to DynamicLoader, in that case would it be messier?